src/snowflake/connector/cursor.py

```diff
@@ -957,7 +968,7 @@ def check_can_use_pandas(self):
                 },
             )
 
-    def query_result(self, qid):
+    def query_result(self, qid: str) -> SnowflakeCursor:
```

Comment:
can use `Self`?
```suggestion
    def query_result(self, qid: str) -> Self:
```

Reply:
We should be able to. `mypy` is supposed to support this through `typing_extensions` for older versions. https://peps.python.org/pep-0673/